tecton_client/_internal/async_tecton_client.py

```diff
@@ -0,0 +1,104 @@
+import json
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Union
+from urllib.parse import urljoin
+
+import httpx
+from httpx import HTTPStatusError
+
+from tecton_client._internal.data_types import GetFeatureServiceMetadataResponse
+from tecton_client._internal.data_types import GetFeaturesResponse
+from tecton_client._internal.utils import build_get_feature_service_metadata_request
+from tecton_client._internal.utils import build_get_features_request
+from tecton_client._internal.utils import get_default_headers
+from tecton_client._internal.utils import validate_request_args
+from tecton_client.exceptions import convert_exception
+
+
+class AsyncTectonClient:
+    """A lightweight http client for interacting with features in Tecton. For the full sdk, use tecton-sdk"""
+
+    def __init__(
+        self, url: str, api_key: str, default_workspace_name: Optional[str] = None, client: httpx.AsyncClient = None
+    ):
+        """Constructor for the client
+
+        Args:
+            url: base url to your tecton cluster. Ex: http://explore.tecton.ai
+            api_key: See https://docs.tecton.ai/docs/ for how to create an api key.
+            default_workspace_name: The workspace from which the features will be retrieved.
+                Can be over-ridden by individual function calls.
+            client: An httpx.Client, allowing you to provide finer-grained customization on the request behavior,
+                such as default timeout or connection settings. See https://www.python-httpx.org/ for more info.
+        """
+        self.url = url
+        self.default_workspace_name = default_workspace_name
+        self._api_key = api_key
+        self._base_url = urljoin(url, "/api/v1/")
+
+        headers = get_default_headers(api_key)
+        if client is None:
+            self._client = httpx.Client(headers=headers)
```

Comment:
AsyncClient?

Reply:
thanks, result of utils-pattern refactor